components/job-orchestration/job_orchestration/executor/compress/fs_compression_task.py

```diff
@@ -76,6 +79,15 @@ def make_clp_s_command(
     return compression_cmd
 
 
+def update_tags(db_conn, db_cursor, table_prefix, archive_id, tags):
+    db_cursor.executemany(
+        f"INSERT INTO {table_prefix}archive_tags (archive_id, tag_id) VALUES "
+        f"(%s, (SELECT tag_id FROM {table_prefix}tags WHERE tag_name = %s))",
```

Comment:
This subquery can still be expensive since we're doing the tag_id lookup, per tag, per archive, right? Instead, if we get the tag IDs in the scheduler and pass them down, we can get rid of the subquery.